packages/web-components/src/components/ic-date-input/ic-date-input.tsx

```diff
@@ -0,0 +1,1132 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Prop,
+  State,
+  Host,
+} from "@stencil/core";
+import {
+  IcDateFormat,
+  IcDateInputMonths,
+  IcDisabledDateTypes,
+} from "./ic-date-input.types";
+import { IcInformationStatusOrEmpty } from "../../interface";
+import { IcInformationStatus } from "../../utils/types";
+import {
+  addFormResetListener,
+  getInputDescribedByText,
+  isEmptyString,
+  removeFormResetListener,
+  renderHiddenInput,
+} from "../../utils/helpers";
+
+let inputIds = 0;
+
+@Component({
+  tag: "ic-date-input",
+  styleUrl: "ic-date-input.css",
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class DateInput {
+  @Element() el: HTMLIcDateInputElement;
+
+  private dayInputEl: HTMLInputElement;
+  private monthInputEl: HTMLInputElement;
+  private yearInputEl: HTMLInputElement;
+  private inputCompContainerEl: HTMLIcInputComponentContainerElement;
+  private selectedDateInfoEl: HTMLSpanElement;
+
+  private screenReaderInfoId: string;
+  private assistiveHintId: string;
+  private selectedDateInfoId: string;
+
+  private previousSelectedDate: Date = null;
+  private selectedDate: Date = null;
+
+  private preventDayInput: boolean;
+  private preventMonthInput: boolean;
+  private preventYearInput: boolean;
+
+  private inputsInOrder: HTMLInputElement[] = [];
+
+  private invalidDateText: string;
+
+  private preventAutoFormatting: boolean;
+
+  private isDateSetFromKeyboardEvent: boolean = false;
+  private initialValue = this.value;
+
+  private FIT_TO_VALUE = "fit-to-value";
+  private EVENT_OBJECT_STRING = "[object Event]";
+  private KEYBOARD_EVENT_OBJECT_STRING = "[object KeyboardEvent]";
+  private INPUT_EVENT_OBJECT_STRING = "[object InputEvent]";
+  private ARIA_INVALID = "aria-invalid";
+  private ARIA_LABEL = "aria-label";
+
+  private isValidDay: boolean = true;
+  private isValidMonth: boolean = true;
+  private isValidDate: boolean = true;
+  private isDisabledDate: boolean = false;
+
+  /**
+   * The ID for the input.
+   */
+  @Prop() inputId?: string = `ic-date-input-${inputIds++}`;
+
+  /**
+   * The label for the date input.
+   */
+  @Prop() label!: string;
+
+  /**
+   * If `true`, the input will require a value.
+   */
+  @Prop() required: boolean = false;
```

Comment:
A few of these props are missing the question mark to show that they're optional